java/adapter/avro/src/main/java/org/apache/arrow/AvroToArrowUtils.java

```diff
@@ -113,7 +113,13 @@ private static ArrowType getArrowType(Type type) {
     schema.getObjectProps().forEach((k,v) -> metadata.put(k, v.toString()));
 
     if (type == Type.RECORD) {
-      throw new UnsupportedOperationException();
+      for (Schema.Field field : schema.getFields()) {
+        final ArrowType arrowType = getArrowType(field.schema().getType());
+        final FieldType fieldType = new FieldType(/*nullable=*/true, arrowType, /*dictionary=*/null, /*metadata=*/null);
```

Comment:
this should be false?

Reply:
fixed. Now false is enough, when union type is supported, this should be true?

Reply:
I think when union type is supported there will need to be a check if the schema is simply a union of [null, type] then the type becomes nullable, if it is more then that it would map to the arrow union type, does that sound right?

Reply:
Sure, exactly 👍